### PR TITLE
chore: retire containertest

### PR DESCRIPTION
**Pull request by mdelapenya**
**Description:** <Describe what has changed.>

It's not used anymore, in favor of testcontainers-go

**Link to tracking Issue:** #9939
- Continuation of #16209
- Continuation of #16264
- Continuation of #16306

**Documentation:** No docs were updated

### CODE REVIEW

File: cmd/configschema/go.mod [hunk 1]
```diff
@@ -670,8 +670,6 @@ replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/x
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/common => ../../internal/common
 
-replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/containertest => ../../internal/containertest
```
Comment (djaglowski) [inline cmd/configschema/go.mod:673]: I believe there is one more reference in the repo's root `go.mod`

File: receiver/mongodbreceiver/actual_metrics.json [hunk 2]
```diff
@@ -0,0 +1,2006 @@
+{
```
Comment (djaglowski) [inline receiver/mongodbreceiver/actual_metrics.json:1]: Is this intentional?

File: cmd/configschema/go.mod (hunk 1 above)
Comment (djaglowski) [reply cmd/configschema/go.mod:673]: Also a reference in `versions.yaml`

File: receiver/mongodbreceiver/actual_metrics.json (hunk 2 above)
Comment (mdelapenya) [reply receiver/mongodbreceiver/actual_metrics.json]: Resolved in [dd1469a](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16316/commits/dd1469ad255ab3d2a931d418b4d626d8714b9ea2)

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Thank you @mdelapenya

**Review by codeboten — APPROVED** [review]
🥳

**Review by djaglowski — APPROVED** [review]

**Pull request merged**